api/indexer/server.go

```diff
@@ -46,11 +46,8 @@ func (f *apiFactory) New(vm api.VM) (api.Handler, error) {
 }
 
 type GetBlockRequest struct {
-	BlockID ids.ID `json:"blockID"`
-}
-
-type GetBlockByHeightRequest struct {
-	Height uint64 `json:"height"`
+	BlockID     ids.ID `json:"blockID"`
+	BlockNumber uint64 `json:"blockNumber"`
```

Comment:
(out of scope for this PR) Can we add an `Encoding` argument that can be either `json` or `hex`, defaults to `hex`, and specifies which field in the response to fill out ie. we populate either the block bytes or block field, but not both?

Can we apply the same to the transaction and either populate the JSON of the transaction or the transaction bytes?

Reply:
yes, I'll tackle this on the next PR.

Reply:
I'd like to have some more unit testing for the server as well.